pkg/internal/approver/manager/predicate/predicate.go

```diff
@@ -77,6 +86,64 @@ func SelectorIssuerRef(_ context.Context, cr *cmapi.CertificateRequest, policies
 	return matchingPolicies, nil
 }
 
+// SelectorNamespace is a Predicate that returns the subset of given policies
+// that have an `spec.selector.namespace` matching the `metadata.namespace` of
+// the request. SelectorNamespace will match with `namespace.matchNames` on
+// namespaces using wilcards "*". Empty selector is equivalent to "*" and will
+// match on any Namespace.
+func SelectorNamespace(lister client.Reader) Predicate {
+	return func(ctx context.Context, cr *cmapi.CertificateRequest, policies []policyapi.CertificateRequestPolicy) ([]policyapi.CertificateRequestPolicy, error) {
+		var matchingPolicies []policyapi.CertificateRequestPolicy
+
+		var namespace corev1.Namespace
+		if err := lister.Get(ctx, client.ObjectKey{Name: cr.Namespace}, &namespace); err != nil {
+			return nil, fmt.Errorf("failed to get request's namespace to determine namespace selector: %w", err)
+		}
+		namespaceLabels := labels.Set(namespace.Labels)
+
+		for _, policy := range policies {
+			nsSel := policy.Spec.Selector.Namespace
+
+			// Namespace Selector is nil so we always match.
+			if nsSel == nil {
+				matchingPolicies = append(matchingPolicies, policy)
+				continue
+			}
+
+			ns := cr.Namespace
+			matched := len(nsSel.MatchNames) == 0
+
+			// Match by name.
+			for _, matchName := range nsSel.MatchNames {
+				if util.WildcardMatchs(matchName, ns) {
```

Comment:
`WildcardMatches` (missing an `e`)

Reply:
Thanks, I'll change this in a different PR to have separate concerns since the function itself wasn't added in this PR.

Reply:
See https://github.com/cert-manager/approver-policy/pull/98